examples/hosts/app-integration/external-data/src/mock-customer-service/service.ts

```diff
@@ -167,6 +174,75 @@ export async function initializeCustomerService(props: ServiceProps): Promise<Se
 		}
 	});
 
+	/**
+	 * Creates an entry in the Customer Service of the mapping between the container and the external resource id
+	 * (externalTaskListId in this example). Also, it signs up the container with the external service
+	 * so that when there is a change upstream and it uses a webhook notification to inform the customer service,
+	 * the customer service can in turn notify the container of the change.
+	 *
+	 * Expected input data format:
+	 *
+	 * ```json
+	 *	{
+	 *		containerUrl: string,
+	 *		externalTaskListId: string
+	 *	}
+	 * ```
+	 *
+	 * Note: Implementers choice --can choose to break up containerUrl into multiple pieces
+	 * containing tenantId, documentId and socketStreamURL separately and send them as a json
+	 * object. The URL also contains all this information so for simplicity I use the url here.
+	 */
+	expressApp.post("/register-session-url", (request, result) => {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+		const containerUrl = request.body?.containerUrl as string;
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+		const externalTaskListId = request.body?.externalTaskListId as string;
+		if (containerUrl === undefined) {
+			const errorMessage =
+				'No session data provided by client. Expected under "sessionUrl" property.';
+			result.status(400).json({ message: errorMessage });
+		} else if (externalTaskListId === undefined) {
+			const errorMessage =
+				'No external task list id provided by client. Expected under "externalTaskListId" property.';
+			result.status(400).json({ message: errorMessage });
```

Comment:
The risk is still there because `result.send()` won't immediately return from the function. I *think* they're also not necessary when we already did something with the `result` (i.e. `result.status(400).json({ message: errorMessage });`). I was thinking of literal `return;` right after setting the 400 response codes. Which also lets us make each check an independent `if`, and the contents of the last `else` can go right after the second undefined check (and that one does need a `result.send()` at the end because it does nothing else with `result`; at least that's my understanding based on another of @Josmithr PRs where I asked about that).